test/ectest.c

```diff
@@ -188,1185 +146,1027 @@ static void group_order_tests(EC_GROUP *group)
-        if (!EC_POINT_get_affine_coordinates_GFp(group, Q, x, y, ctx))
-            ABORT;
-        fprintf(stderr, "Point is not on curve: x = 0x");
+    if (!TEST_ptr(P = EC_POINT_new(group)) || !TEST_ptr(Q = EC_POINT_new(group))
+        || !TEST_ptr(R = EC_POINT_new(group))
+        || !TEST_true(EC_POINT_set_to_infinity(group, P))
+        || !TEST_true(EC_POINT_is_at_infinity(group, P))
+        || !TEST_true(EC_POINT_oct2point(group, Q, buf, 1, ctx))
+        || !TEST_true(EC_POINT_add(group, P, P, Q, ctx))
+        || !TEST_true(EC_POINT_is_at_infinity(group, P))
+        || !TEST_ptr(x = BN_new()) || !TEST_ptr(y = BN_new())
+        || !TEST_ptr(z = BN_new()) || !TEST_ptr(yplusone = BN_new())
+        || !TEST_true(BN_hex2bn(&x, "D"))
+        || !TEST_true(EC_POINT_set_compressed_coordinates_GFp(group, Q, x, 1,
+                                                              ctx)))
+        goto err;
+
+    if (!TEST_int_gt(EC_POINT_is_on_curve(group, Q, ctx), 0)) {
+        if (!TEST_true(EC_POINT_get_affine_coordinates_GFp(group, Q, x, y,
+                                                           ctx)))
+            goto err;
+        BIO_printf(bio_err, "Point is not on curve: x = 0x");
         BN_print_fp(stderr, x);
-        fprintf(stderr, ", y = 0x");
+        BIO_printf(bio_err, ", y = 0x");
         BN_print_fp(stderr, y);
-        fprintf(stderr, "\n");
-        ABORT;
+        BIO_printf(bio_err, "\n");
+        goto err;
     }
 
-    fprintf(stdout, "A cyclic subgroup:\n");
+    BIO_printf(bio_out, "A cyclic subgroup:\n");
     k = 100;
     do {
-        if (k-- == 0)
-            ABORT;
-
-        if (EC_POINT_is_at_infinity(group, P))
-            fprintf(stdout, "     point at infinity\n");
-        else {
-            if (!EC_POINT_get_affine_coordinates_GFp(group, P, x, y, ctx))
-                ABORT;
-
-            fprintf(stdout, "     x = 0x");
-            BN_print_fp(stdout, x);
-            fprintf(stdout, ", y = 0x");
-            BN_print_fp(stdout, y);
-            fprintf(stdout, "\n");
+        if (!TEST_int_ne(k--, 0))
+            goto err;
+
+        if (EC_POINT_is_at_infinity(group, P)) {
+            BIO_printf(bio_out, "     point at infinity\n");
+        } else {
+            if (!TEST_true(EC_POINT_get_affine_coordinates_GFp(group, P, x, y,
+                                                               ctx)))
+                goto err;
+
+            BIO_printf(bio_out, "     x = 0x");
+            BN_print(bio_out, x);
+            BIO_printf(bio_out, ", y = 0x");
+            BN_print(bio_out, y);
+            BIO_printf(bio_out, "\n");
         }
 
-        if (!EC_POINT_copy(R, P))
-            ABORT;
-        if (!EC_POINT_add(group, P, P, Q, ctx))
-            ABORT;
+        if (!TEST_true(EC_POINT_copy(R, P))
+            || !TEST_true(EC_POINT_add(group, P, P, Q, ctx)))
+            goto err;
 
     }
     while (!EC_POINT_is_at_infinity(group, P));
```

Comment:
"while" you're editing this file, put the while up with the curly brace at 250/251.